envoy/network/address.h

```diff
@@ -230,6 +231,32 @@ class Instance {
    * @return SocketInterface to be used with the address.
    */
   virtual const Network::SocketInterface& socketInterface() const PURE;
+
+  /**
+   * For support the absl::Hash.
```

Comment:
-> For supporting absl::Hash

Reply:
Instance::abslHashValue's documentation should say something like:
// Overrides must ensure that everything that affects operator== is combined into state, and vice-versa.
// See https://abseil.io/docs/cpp/guides/hash#making-hashable-types.

Reply:
got it, thanks!

Reply:
> Instance::abslHashValue's documentation should say something like: // Overrides must ensure that everything that affects operator== is combined into state, and vice-versa. // See https://abseil.io/docs/cpp/guides/hash#making-hashable-types.

I need to use `InstanceConstSharedPtr` as the hash key. Not the actual address object. Then I added `struct AddressKey` below, the `operator==` is defined for AddressKey also.